### PR TITLE
Bug layer ordering reversed

### LINKED ISSUE #189: Bug: Layer ordering can end up in a reversed state
A bug surfaced that causes the property order to be set incorrectly. In normal operation, this bug was resolved by a double negative, that is, a situation where the ordered array was reversed, and then reversed again later.

However, the health check endpoint calls a getter which mutates the ordered array in place, causing this bug to occur.

The list of active sources is reversed by the [`Properties.sources()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/properties.js#L34) getter. That getter is called by the [`Sources.health()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/sources.js#L155) function used by both the `/v1/health` and `/v1/status` APIs.

Additionally, the list of layers is reversed every time [`Properties.build()`](https://github.com/rapid7/propsd/blob/v1.3.1/lib/properties.js#L117) is called.

### PR DESCRIPTION
**Pull request by dgreene-r7**
This PR refactors the `Properties` class to never reverse internal arrays. This fixes problems where the `Properties#sources` getter and `Properties#build()` were mutating objects in place. It reverses the order in which Consul, EC2 Metadata, and static properties are loaded in `server.js`.

Resolves #189 and supersedes #192 and #196.

### CODE REVIEW

File: lib/properties.js [hunk 1]
```diff
@@ -29,9 +29,8 @@ class Properties extends EventEmitter {
       // Get Dynamic layers' source instances
       .concat(this.layers.map((layer) => layer.source).filter((source) => !!source))
 
-      // Add the active View's sources. Reversed for precedence consistency with Layers:
-      // Entry 0 is most precedent, LAST is least precedent.
-      .concat(this.active.sources.reverse());
+      // Add the active View's sources.
+      .concat(this.active.sources);
```
Comment (fmitchell-r7) [inline lib/properties.js]: Do we want to keep the API unchanged for `/v1/status` and have this go out as a bug fix? If so, we can write this as
```javascript
return []
  .concat(this.active.sources)
  .concat(this.layers.map((layer) => layer.source).filter((source) => !!source))
  .reverse();
```

Comment (dgreene-r7) [inline lib/properties.js]: If we want to keep it unchanged we'd do:
```javascript
return []
  .concat(this.layers.map((layer) => layer.source).filter((source) => !!source))
  .concat(this.active.sources.slice().reverse());
```

Keep in mind the reversal was applying to the order in which active sources were listed in the `/v1/status` endpoint so we have to reverse `this.active.sources` before we `concat` it with the rest of the sources.

I'm ambivalent. Let me know your preference and we can do it however you want.

Comment (fmitchell-r7) [inline lib/properties.js]: Let's keep the API unchanged and have this go out as a bug fix.

File: test/properties.js [hunk 2]
```diff
@@ -34,6 +34,55 @@ describe('Properties', function _() {
...
+
+  it('does not reorder layers if the properties sources getter is called', function ___(done) {
+    const props = new Properties();
+    const sources = [
+      new Source.Stub({path: 'foo'}),
+      new Source.Stub({path: 'bar'}),
+      new Source.Stub({path: 'baz'})
+    ];
+    const view = props.view(sources);
+    const expected = ['baz', 'bar', 'foo'];
```
Comment (fmitchell-r7) [inline test/properties.js]: I can't get this test to fail correctly against the v1.4.2 code line. Try this instead?
```javascript
const expected = ['foo', 'bar', 'baz'];
const reversed = ['baz', 'bar', 'foo'];

props.once('build', () => {
  expect(props.sources.map((s) => s.properties.path)).to.eql(reversed);
  expect(props.active.sources.map((s) => s.properties.path)).to.eql(expected);
  done();
});

view.activate();
```

The value returned from `props.sources` _should_ be reversed, but the value returned from `props.active.sources` _should not_ be reversed.

**Comment by fmitchell-r7** [comment]
👍  Thanks for fixing this! 🚢  it.

**Pull request merged**